collectd/_service.sls

```diff
@@ -129,6 +129,36 @@
 
 {%- endif %}
 
+{%- set java_lib = salt['cmd.run']('type java >/dev/null 2>&1 && find /usr/lib/jvm -name libjvm.so -type f') %}
+{%- if java_lib|length > 0 %}
+  {%- set java_path = '/'.join(java_lib.split('/')[:-1]) %}
+  {%- set ld_is_set = salt['cmd.run']('grep "^export LD_LIBRARY_PATH" ' ~ client.defaults_file ~ ' 2>/dev/null') %}
+  {%- if ld_is_set|length > 0 %}
+    {%- set ld_paths = ld_is_set.split('=')[1].split(':') %}
+    {%- if java_path not in ld_paths %}
+      {%- set ld_mod_text = java_path %}
+    {%- endif %}
+  {%- else %}
+    {%- set ld_add_text = 'export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:' ~ java_path %}
```

Comment:
I guess **exporting** LD_LIBRARY_PATH is required to make it available to the `java` subprocess?